Index: control/matlab/QPController.m
===================================================================
--- control/matlab/QPController.m	(revision 3369)
+++ control/matlab/QPController.m	(working copy)
@@ -167,13 +167,16 @@
         obj.rfoot_contact_state = msg.right_contact;
       end
     else
-      contact_threshold = 0.002; % m
+      contact_threshold = 0.003; % m
       q = x(1:nq); 
       kinsol = doKinematics(r,q,false,true);
     
       % get active contacts
       phi = contactConstraints(r,kinsol,[obj.lfoot_idx,obj.rfoot_idx]);
 
+      % DEBUG
+      phi
+      
       % if any foot point is in contact, all contact points are active
       if any(phi(1:4)<contact_threshold)
         obj.lfoot_contact_state = 1;
@@ -473,9 +476,11 @@
       else
         xcomdd = Jdot * qd + J * alpha(1:nq);
       end
-      zmppos = xcom(1:2) + D_ls * xcomdd;
-      % Set zmp z-pos to 1m for DRC Quals 1
-      plot_lcm_points([zmppos', terrain_height], [1, 0, 0], 660, 'Commanded ZMP', 1, true);
+%       zmppos = xcom(1:2) + D_ls * xcomdd;
+%       % Set zmp z-pos to 1m for DRC Quals 1
+%       plot_lcm_points([zmppos', terrain_height], [1, 0, 0], 660, 'Commanded ZMP', 1, true);
+      xcom_nom = ctrl_data.comtraj.eval(t);
+      plot_lcm_points([xcom_nom', terrain_height; xcom(1:2)', terrain_height], [1, 0, 0; 0 1 0], 660, 'Commanded ZMP', 1, true);
       
       [cheight,normals] = getTerrainHeight(r,cpos);
       d = RigidBodyManipulator.surfaceTangents(normals);
