Index: xacro/inert_hand_left.urdf.xacro
===================================================================
--- xacro/inert_hand_left.urdf.xacro	(revision 8197)
+++ xacro/inert_hand_left.urdf.xacro	(working copy)
@@ -12,8 +12,8 @@
     </inertial>
   </link>
   
-<!-- <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />  -->
- <xacro:include filename="$(find hook_description)/src/hook_hand_simple.urdf.xacro" />
+ <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />  
+ <!--<xacro:include filename="$(find hook_description)/src/hook_hand_simple.urdf.xacro" />-->
 
   <xacro:hook_hand side="left" parent="plane::xy::base" reflect="-1" hookflip="-1" noof30degoffset="0">
     <origin rpy="1.57 0.0 -1.57" xyz="0.243 0.0 -0.127"/>
Index: xacro/inert_hand_right.urdf.xacro
===================================================================
--- xacro/inert_hand_right.urdf.xacro	(revision 8197)
+++ xacro/inert_hand_right.urdf.xacro	(working copy)
@@ -12,8 +12,8 @@
     </inertial>
   </link>
   
-<!--  <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />  -->
-  <xacro:include filename="$(find hook_description)/src/hook_hand_simple.urdf.xacro" />
+  <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />  
+  <!--<xacro:include filename="$(find hook_description)/src/hook_hand_simple.urdf.xacro" />-->
 
    <xacro:hook_hand side="right" parent="plane::xy::base" reflect="1" hookflip="-1" noof30degoffset="0">
      <origin rpy="1.57 3.14 -1.57" xyz="0.243 0.0 -0.127"/>
