Index: mit_robot_hands/inert_hand_right.urdf
===================================================================
--- mit_robot_hands/inert_hand_right.urdf	(revision 7796)
+++ mit_robot_hands/inert_hand_right.urdf	(working copy)
@@ -26,13 +26,13 @@
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="0.001 0.001 0.001"/>
       </geometry>
-      <origin rpy="0 0 3.14" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
     </collision>
     <visual>
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="1 1 1"/>
       </geometry>
-      <origin rpy="0 0 3.14" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
     </visual>
   </link>
 </robot>
Index: mit_robot_hands/xacro/inert_hand_left.urdf.xacro
===================================================================
--- mit_robot_hands/xacro/inert_hand_left.urdf.xacro	(revision 7796)
+++ mit_robot_hands/xacro/inert_hand_left.urdf.xacro	(working copy)
@@ -13,7 +13,7 @@
   </link>
   
  <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />
-  <xacro:hook_hand side="left" parent="plane::xy::base" reflect="-1" hookflip="1">
+  <xacro:hook_hand side="left" parent="plane::xy::base" reflect="-1" hookflip="-1">
     <origin rpy="1.57 0.0 -1.57" xyz="0.0 0.0 0.0"/>
  </xacro:hook_hand>
 </robot>
Index: mit_robot_hands/xacro/inert_hand_right.urdf.xacro
===================================================================
--- mit_robot_hands/xacro/inert_hand_right.urdf.xacro	(revision 7796)
+++ mit_robot_hands/xacro/inert_hand_right.urdf.xacro	(working copy)
@@ -13,7 +13,7 @@
   </link>
   
   <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />
-   <xacro:hook_hand side="right" parent="plane::xy::base" reflect="1" hookflip="1">
+   <xacro:hook_hand side="right" parent="plane::xy::base" reflect="1" hookflip="-1">
      <origin rpy="1.57 3.14 -1.57" xyz="0 0 0"/>
   </xacro:hook_hand>  
 </robot>
Index: mit_robot_hands/inert_hand_left.urdf
===================================================================
--- mit_robot_hands/inert_hand_left.urdf	(revision 7796)
+++ mit_robot_hands/inert_hand_left.urdf	(working copy)
@@ -26,13 +26,13 @@
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="0.001 0.001 0.001"/>
       </geometry>
-      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 -3.14" xyz="0.00 0.0 0.0"/>
     </collision>
     <visual>
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="1 1 1"/>
       </geometry>
-      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 -3.14" xyz="0.00 0.0 0.0"/>
     </visual>
   </link>
 </robot>
Index: mit_robot/model_LH_RH.urdf
===================================================================
--- mit_robot/model_LH_RH.urdf	(revision 7796)
+++ mit_robot/model_LH_RH.urdf	(working copy)
@@ -1368,13 +1368,13 @@
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="0.001 0.001 0.001"/>
       </geometry>
-      <origin rpy="0 0 3.14" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
     </collision>
     <visual>
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="1 1 1"/>
       </geometry>
-      <origin rpy="0 0 3.14" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
     </visual>
   </link>
   <joint name="left_hook_hand_joint" type="fixed">
@@ -1392,13 +1392,13 @@
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="0.001 0.001 0.001"/>
       </geometry>
-      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 -3.14" xyz="0.00 0.0 0.0"/>
     </collision>
     <visual>
       <geometry>
         <mesh filename="package://hook_description/meshes/bdi_hook.stl" scale="1 1 1"/>
       </geometry>
-      <origin rpy="0 0 0.0" xyz="0.00 0.0 0.0"/>
+      <origin rpy="0 0 -3.14" xyz="0.00 0.0 0.0"/>
     </visual>
   </link>
 </robot>
Index: mit_robot/xacro/atlas_LH_RH.urdf.xacro
===================================================================
--- mit_robot/xacro/atlas_LH_RH.urdf.xacro	(revision 7796)
+++ mit_robot/xacro/atlas_LH_RH.urdf.xacro	(working copy)
@@ -6,10 +6,10 @@
 
  <xacro:include filename="$(find hook_description)/src/bdihook_hand.urdf.xacro" />
 
-  <xacro:hook_hand side="right" parent="r_hand" reflect="1" hookflip="1">
+  <xacro:hook_hand side="right" parent="r_hand" reflect="1" hookflip="-1">
     <origin rpy="1.57079 0 0" xyz="0 -0.11516 -0.015"/>
   </xacro:hook_hand>
-  <xacro:hook_hand side="left" parent="l_hand" reflect="-1" hookflip="1">
+  <xacro:hook_hand side="left" parent="l_hand" reflect="-1" hookflip="-1">
     <origin rpy="1.57079 3.14159 3.14159" xyz="0 0.11516 0.015"/>
   </xacro:hook_hand>
 
