Index: kinect/libfreenect/examples/CMakeLists.txt
===================================================================
--- kinect.orig/libfreenect/examples/CMakeLists.txt	2010-12-17 00:47:23.000000000 -0500
+++ kinect/libfreenect/examples/CMakeLists.txt	2010-12-17 00:47:34.000000000 -0500
@@ -19,7 +19,8 @@
 else()
   find_package(Threads REQUIRED)
   find_package(OpenGL REQUIRED)
-  find_package(GLUT REQUIRED)
+  set(GLUT_INCLUDE_DIRS)
+  set(GLUT_LIBRARY -lglut)
   include_directories(${OPENGL_INCLUDE_DIRS} ${GLUT_INCLUDE_DIRS} ${USB_INCLUDE_DIRS})
   target_link_libraries(glview freenect ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${CMAKE_THREAD_LIBS_INIT} m)
   target_link_libraries(cppview freenect ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${CMAKE_THREAD_LIBS_INIT} m)
